### PR TITLE
Update missing-owner-check lint to consider anchor constraints applied on the accounts

### PR DESCRIPTION
**Pull request by S3v3ru5**
The lint now checks if the AccountInfo expression accesses the field of a struct having `#[derive(Accounts)]` (Anchor Accounts structs). If so the lint checks for constraints applied on the field using `#[account(...)]`. The lint will *ignore* the expression if any of the constraints are applied on the field:

- `#[account(signer)]` - Signer accounts are assumed to be EOA accounts and are ignored.
- `#[account(init, ...)]` - init creates a new account and sets its owner to the current program or the given program.
- `#[account(seeds = ..., ...)]` - Anchor derives a PDA using the seeds. This is essentially a `key` check
- `#[account(address = ...)]` - Validates account's key.
- `#[account(owner = ...)]` - Validates the owner.
- `#[account(executable)]` - The account is an executable; All executables are owned by `BPFLoaders`.

### CODE REVIEW

File: lints/missing_owner_check/Cargo.toml [hunk 1]
```diff
@@ -33,18 +33,33 @@ path = "ui/secure-account-owner/src/lib.rs"
...
 solana-lints = { path = "../../crate" }
+syn = { version = "1.0.109", features = ["parsing"] }
 
 [dev-dependencies]
 anchor-lang = "0.29"
 anchor-spl = "0.29"
 dylint_testing = "2.6"
 
 [workspace]
+# members = [
```
Comment (S3v3ru5) [inline lints/missing_owner_check/Cargo.toml:54]: Missed these comments. Will remove along with the fix to doc-tests.

**Review by smoelius — COMMENTED** [review]
One question. Everything else looks okay.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 /target
 /lints/*/ui/*/Cargo.lock
+# /lints/debug/*
+/lints/*/ui/Cargo.toml
```
[the remark this answers is not in the document]
Comment (S3v3ru5) [reply .gitignore:4]: Sorry, I thought I removed this. I added them while testing. I will remove them. Thanks for pointing out.

**Comment by S3v3ru5** [comment]
`npm install -g markdown-link-check && find . -name '*.md' | xargs markdown-link-check`

^running above command is failing the `CI / lint` action. I think the problem is with the `markdown-link-check` library. I tried running the `markdown-link-check` on an empty `.md` file and it gave the same error.

And also, I only removed the unnecessary lines from the `.gitignore` in the last commit. The actions have all passed before that.

**Pull request merged**